pymc/logprob/binary.py

```diff
@@ -50,26 +51,35 @@ def find_measurable_comparisons(
     if isinstance(node.op, MeasurableComparison):
         return None  # pragma: no cover
 
-    (compared_var,) = node.outputs
-    base_var, const = node.inputs
+    measurable_inputs = [
+        inp
+        for idx, inp in enumerate(node.inputs)
+        if inp.owner
+        and isinstance(inp.owner.op, MeasurableVariable)
+        and inp not in rv_map_feature.rv_values
+    ]
 
-    if not (
-        base_var.owner
-        and isinstance(base_var.owner.op, MeasurableVariable)
-        and base_var not in rv_map_feature.rv_values
-    ):
+    if len(measurable_inputs) != 1:
         return None
 
+    base_var: TensorVariable = measurable_inputs[0]
+
+    # Check that the other input is not potentially measurable, in which case this rewrite
+    # would be invalid
+    const = tuple(inp for inp in node.inputs if inp is not base_var)
+
     # check for potential measurability of const
-    if not check_potential_measurability((const,), rv_map_feature):
+    if not check_potential_measurability(const, rv_map_feature):
         return None
 
+    const = const[0]
+
     # Make base_var unmeasurable
     unmeasurable_base_var = ignore_logprob(base_var)
 
     compared_op = MeasurableComparison(node.op.scalar_op)
     compared_rv = compared_op.make_node(unmeasurable_base_var, const).default_output()
```

Comment:
Won't you have to change the `scalar_op` used depending on the order? 

If you have `gt(const, dist)` you don't want to return `measurable_gt(dist, const)`, you changed the meaning!

Reply:
Oh you're absolutely right! I missed it while thinking about the structure of the rewrite. Will make the change right away.